### DISCUSSION TITLE
How can created associated Token Account for SOLANA NFT

### DISCUSSION THREAD

**Discussion by blukhi99** [trustwallet/wallet-core#2850] [Q&A]
I am trying to create and transfer Solana NFT to another wallet.
mostly NFTs are working well transfer and create an associated account if it is not available.
but, no luck for one NFT.

below is my NFT code and **AxqWgkgAkqycRzsdAseUY4DwSRpKXgC9Gaok7EoVBmVN** this is my NFT address.

can any one help me with what I am getting wrong here? 

       ` val _privateKey = wallet.getPrivateKeyAsBytes(CoinType.SOLANA.value())
        val transferMessage = Solana.CreateAndTransferToken.newBuilder().apply {
            this.recipientMainAddress = receiver
            this.tokenMintAddress = contractAddress
            this.senderTokenAddress = senderTokenAddress
            this.recipientTokenAddress =
                associatedAddress(receiver, contractAddress = contractAddress)
            this.amount = amount.toBigDecimal().toWEI(decimals).toLong()
            this.decimals = decimals
        }.build()
        val signingInput = Solana.SigningInput.newBuilder().apply {
            this.privateKey = ByteString.copyFrom(_privateKey)
            this.createAndTransferTokenTransaction = transferMessage
            this.recentBlockhash = blockhash
        }.build()
        return AnySigner.sign(signingInput, CoinType.SOLANA, Solana.SigningOutput.parser())`

Error is

`{"jsonrpc":"2.0","error":{"code":-32002,"message":"Transaction simulation failed: Error processing Instruction 1: invalid account data for instruction","data":{"accounts":null,"err":{"InstructionError":[1,"InvalidAccountData"]},"logs":["Program ATokenGPvbdGVxr1b2hvZbsiqW5xWH25efTNsLJA8knL invoke [1]","Program log: Create","Program TokenkegQfeZyiNwAJbNbGKPFXCWuBvf9Ss623VQ5DA invoke [2]","Program log: Instruction: GetAccountDataSize","Program TokenkegQfeZyiNwAJbNbGKPFXCWuBvf9Ss623VQ5DA consumed 1622 of 394408 compute units","Program return: TokenkegQfeZyiNwAJbNbGKPFXCWuBvf9Ss623VQ5DA pQAAAAAAAAA=","Program TokenkegQfeZyiNwAJbNbGKPFXCWuBvf9Ss623VQ5DA success","Program 11111111111111111111111111111111 invoke [2]","Program 11111111111111111111111111111111 success","Program log: Initialize the associated token account","Program TokenkegQfeZyiNwAJbNbGKPFXCWuBvf9Ss623VQ5DA invoke [2]","Program log: Instruction: InitializeImmutableOwner","Program log: Please upgrade to SPL Token 2022 for immutable owner support","Program TokenkegQfeZyiNwAJbNbGKPFXCWuBvf9Ss623VQ5DA consumed 1405 of 387918 compute units","Program TokenkegQfeZyiNwAJbNbGKPFXCWuBvf9Ss623VQ5DA success","Program TokenkegQfeZyiNwAJbNbGKPFXCWuBvf9Ss623VQ5DA invoke [2]","Program log: Instruction: InitializeAccount3","Program TokenkegQfeZyiNwAJbNbGKPFXCWuBvf9Ss623VQ5DA consumed 4241 of 384034 compute units","Program TokenkegQfeZyiNwAJbNbGKPFXCWuBvf9Ss623VQ5DA success","Program ATokenGPvbdGVxr1b2hvZbsiqW5xWH25efTNsLJA8knL consumed 20545 of 400000 compute units","Program ATokenGPvbdGVxr1b2hvZbsiqW5xWH25efTNsLJA8knL success","Program TokenkegQfeZyiNwAJbNbGKPFXCWuBvf9Ss623VQ5DA invoke [1]","Program log: Instruction: TransferChecked","Program log: Error: InvalidAccountData","Program TokenkegQfeZyiNwAJbNbGKPFXCWuBvf9Ss623VQ5DA consumed 1455 of 379455 compute units","Program TokenkegQfeZyiNwAJbNbGKPFXCWuBvf9Ss623VQ5DA failed: invalid account data for instruction"],"returnData":null,"unitsConsumed":20545}},"id":1}`

**Comment by KevinC567** [comment]
I also encountered the same problem. Did the author solve this problem? Share the solution

**Reply by satoshiotomakan** [reply]
Hi @KevinC567, please make sure if you use `solanaAddress.defaultTokenAddress(tokenMintAddress)` as the `CreateAndTransferToken.recipientTokenAddress`
